rachellegelden/Home-Together-Canada
# (3/4) HTC-129: created a route to request a new business user

# [HTC-129](https://github.com/rachellegelden/Home-Together-Canada/issues/129)

## Summary
PR Plan for #129 
- ~~(1/4): creating business account model and the controllers to create a business account~~
- ~~(2/4): adding the crypto dependency for password hashing~~
- **(3/4): creating the route and functions to register a business (create an entry in both AbstractUser and BusinessAccount tables)**
- (4/4): adding validation to all fields before creating entries in the AbstractUser and BusinessAccount tables

This PR includes: 
- a refactor of existing controllers
- a route to create a new business user (creates an entry in both the AbstractUser and BusinessAccount table)
- moved `accountControllerUtils` to a `utils` folder
## Relevant Motivation & Context
This is the request that will be used to create a business user from the react app

## Testing Instructions
Using Postman, send a request to `/businessUser/create/` with the following request body:
```
{
    "username": "rachellegelden",
    "password": "password",
    "email": "rachelle@gelden.com",
    "firstName": "Rachelle",
    "lastName": "Gelden",
    "phoneNumber": "(111)-111-1111",
    "addressLine1": "1 Oak Ave",
    "city": "Kelowna",
    "province": "British Columbia",
    "postalCode": "T1V1S4",
    "hasDifferentMailingAddress": false,
    "businessName": "Bob's Bricks",
    "isIncorporated": false,
    "businessPhoneNumber": "(333)-333-3333",
    "businessCellPhoneNumber": "(444)-444-4444",
    "isNationWide": false,
    "mapAddressLine1": "3 Business Drive",
    "mapCity": "Calgary",
    "mapProvince": "Alberta",
    "mapPostalCode": "C3O 2W2",
    "mapLatitude": 51.0447,
    "mapLongitude": 114.0719,
    "website": "www.google.com"
}
```
You should receive a JSON with the abstract user and the business account as a response. Then using the `/businessAccount/all/` and `/abstractUser/all/` check that entries for your abstract user and business account exist.

## Developer checklist prior to opening this pull request:


## Reviewer
